modules/editor/webhook/WebhookControls.tsx

```diff
@@ -68,38 +61,34 @@ export function WebhookControls(props: WebhookControlsProps) {
     }
   }, [])
 
-  return useObserver(() => (
-    <Stack gap={12}>
-      <InputField
-        ref={inputRef}
-        id="webhook"
-        type="password"
-        label="Webhook URL"
-        placeholder="https://discord.com/api/webhooks/..."
-        error={form.subForm("target").field("url").error}
-        {...form.subForm("target").field("url").inputProps}
-      >
-        <PrimaryButton
-          disabled={!editorManager.target.exists}
-          onClick={handleSend}
+  return useObserver(() => {
+    let saveLabel: string
+    if (editorManager.messages.every(m => !m.reference)) saveLabel = "Send"
+    else if (editorManager.messages.every(m => m.reference)) saveLabel = "Edit"
+    else saveLabel = "Submit"
```

Comment:
Maybe initialise saveLabel, as a statement after a bare 'else' looks weird stylistically
```suggestion
    let saveLabel = "Submit"
    if (editorManager.messages.every(m => !m.reference)) saveLabel = "Send"
    else if (editorManager.messages.every(m => m.reference)) saveLabel = "Edit"
```